### PR TITLE
Tutanota blocks public proxies

### PR DESCRIPTION
**Pull request by divinity76**
thus it can be argued that they require your real IP address to sign up..

### CODE REVIEW

**Comment by Admonstrator** [comment]
> thus it can be argued that they require your real IP address to sign up..

It might be an old thing because it's possible now.
Checked it by using Mullvad as my main VPN proxy.

**Review by Lissy93 — COMMENTED** [review]
The IP you're visiting from will always be visible to any service (it's just how servers work), if they're filtering IPs, that doesn't necessarily mean that they're logging that info too.

I'd be interested to learn more about how you were blocked though, as that would fall under censorship, which should be listed as a disadvantage here.

**Comment by divinity76** [comment]
@Lissy93 I was unable to create a Tutanota account trough [Tor](https://www.torproject.org/).

**Comment by Admonstrator** [comment]
> @Lissy93 I was unable to create a Tutanota account trough [Tor](https://www.torproject.org/). (So I signed up with [skiff.com](https://skiff.com/) instead.)
> 
> * iirc i just got a generic error during the signup process

I'll ask the Tuta team about it.

**Comment by kibhub** [comment]
Hi everyone. I'm from the Support team at Tutanota.

TL;DR: We do not store Singup IP addresses nor Login IP addresses. 

This commit lists the IP address as required personal information, which is not the case. We do check the IP against our own list of IP signup reputation during registration but it is not stored in relation to the account created. If the created account needs manual approval, we store an anonymised IP (last one byte) **only until the approval status of the account is confirmed** (which takes a couple of days normally) and then we automatically update our IP signup reputation list. 

Therefore some IP addresses could be either blocked or marked as suspicious, due to signups in the past, but we do not generally target VPNs or networks like Tor.

I hope this clarifies the issue.

**Comment by divinity76** [comment]
Well I just tired again today (2023-10-13), Now I get a specific error message (iirc last time it was a generic one)

>Registration is temporarily blocked for your IP address to avoid abuse. Please try again later or use a different internet connection.

image

not sure what to do with this PR then?

**Comment by Lissy93** [comment]
That's really interesting. Totally makes sense why they'd be doing this (to prevent bot and spam sign-ups).

If they were to use their IP filtering system to prevent signups from certain regions, that would fall under censorship, which should be logged here. But (from what I understand) that doesn't seem to be the case.

Given the reply from @kibhub - I'm going to go ahead and close this PR (as no evidence that Tutanota is logging IP addresses).

---

On a side note, whatever email provider you use, they **WILL have access to your IP address**. So to some extent you do just have to trust they they're not logging this. Even if their code is open source, there's no guarantee that IPs are not logged by the server. If hiding your IP is important to you, you should use a reputable VPN, the Tor network, or similar service.
